### PR TITLE
Add cudf JNI docker build github action

### PR DESCRIPTION
**Pull request by pxLi**
Signed-off-by: Peixin Li <pxli@nyu.edu>

related to https://github.com/NVIDIA/spark-rapids-jni/issues/203

I don't know if any restriction of using github actions in rapids org, 
so this PR is just a prototype and waiting for feedback :)

Example in my forked repo,
1. Manual trigger the build
image
2. example logs https://github.com/pxLi/cudf/runs/6319840419?check_suite_focus=true
3. pushed images https://hub.docker.com/r/pxli/cudf-jni-build/tags
image

Some open questions:
1. Trigger strategy? Manual, crontab, merge trigger, or others?
2. which org in docker hub should we push the image to? gpuci?
3. which account we should use to push?
4. do we want to keep the commit tags or we just overwrite the branch-XY tag only? mostly about space :)

Thanks!

### CODE REVIEW

File: .github/workflows/jni-docker-build.yml
Comment (jlowe) [inline .github/workflows/jni-docker-build.yml:1]: Missing copyrights

Comment (pxLi) [reply .github/workflows/jni-docker-build.yml:1]: I was trying to follow existing cudf workflow files https://github.com/rapidsai/cudf/tree/branch-22.06/.github/workflows
not sure if we need to add the license header here

Comment (pxLi) [reply .github/workflows/jni-docker-build.yml:15]: added

**Comment by pxLi** [comment]
@ajschmidt8 thanks! Just updated the workflow file based on your comments

@jlowe do you think manual trigger is good enough in our case? This is should be not that expensive as cron trigger or post-merge check for all merged commits.
Like if a jni dockerfile change got merged(not that frequent), then the JAVA-code owner should manually trigger the action to do a one-time build

**Review by jlowe — APPROVED** [review]

**Comment by pxLi** [comment]
I will verify the action after merge.

@jlowe sure, we can follow this requirement in  https://github.com/NVIDIA/spark-rapids-jni/issues/203

**Comment by pxLi** [comment]
@gpucibot merge

**Pull request merged**
